explainer.md

```diff
@@ -0,0 +1,255 @@
+::shared-container(#shared-header) {
+  width: 200px;
+  height: 200px;
+  transform: translate(0px, 100px);
+}
+```
+
+7. Once no pseudo element has an active animation, remove them from the top layer.
+
+## Live Animatable Properties
+A common capability desirable during transitions is to interpolate styles like border-radius that change the element's shape. The [container transform](https://material.io/design/motion/the-motion-system.html#container-transform) examples show a visual demo of that. Painting properties like the element's border within its image when using the element() function makes this difficult. The following example[^2] shows how developers can control which styles are captured in the element() function vs being applied to the pseudo container on the old page.
+
+```
+<html>
+<head>
+  <style>
+    .animated {
+      width:100px;
+      height:100px;
+      border: 10px solid black;
+      border-radius: 10% 10%;
+      box-shadow: 0px 0px 10px;
+      
+      &:transition {
+        /* Retain the border to ensure it is painted transparent but the box size is unchanged. */
+        border: 10px solid transparent;
+        border-radius: none;
+        box-shadow: none;
+      }
+    }
+    
+    ::shared-container(#shared-header) {
+      border: 10px solid black;
+      border-radius: 10% 10%;
+      box-shadow: 0px 0px 10px;
+    }
+    
+    ::shared-old(#shared-header) {
+      top: -10px;
+      left: -10px;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header">hello world</div>
+</body>
+</html>
+```
+
+This capability requires the following changes to the steps mentioned in the design above :
+* A new pseudo class ("transition") is introduced which is activated when an element is being displayed using pseudo elements in step 1 and 5.
+* When saving state in step 3, the container element's complete style is preserved instead of only the computed size and transform.
+
+## Modifications to element()
+The following changes will be made to the element() spec as a part of this proposal :
+
+* The target element must have paint containment (contain:paint) to ensure the element is the containing block for all positioned descendents and generates a stacking context.
+* The target element must disallow fragmentation (break-inside:avoid).
+* A new cached-element() function is introduced to refer to the saved output of the element() function in step 3) of the design.
+
+# API Examples
+## SPA
+TODO: Add example
+
+## MPA Script Based
+TODO: Add example
+
+# Alternatives Considered
+## Heirarchical Properties
+This proposal disallows a shared element to be nested inside another shared element. The restriction avoids the need to preserve the hierarchy of the shared elements and associated properties (transform, clip, effects inherited by descendents) when creating pseudo elements. This is a consideration for future iterations of the feature.
```

Comment:
It does not allow nesting? I thought we were allowing that? It's the pseudos that don't nest in the top layer, not the shared elements right?

Reply:
There was a discussion about this in the context of how we'll add the heirarchical aspect incrementally later. Disallowing it for now makes it easy to avoid backward compatibility issues if we change the behaviour to retain the hierarchy in pseudo elements then.